docs/pages/guides/authentication.md

```diff
@@ -109,6 +110,233 @@ export default function App() {
 
 <!-- End IdentityServer 4 -->
 
+### Auth0
+
+<CreateAppButton name="Auth0" href="https://auth0.com/signup" />
+
+| Website                           | Provider | PKCE      | Auto Discovery |
+| --------------------------------- | -------- | --------- | -------------- |
+| [Sign-up][c-auth0] > Applications | OpenID   | Supported | Available      |
+
+[c-auth0]: https://auth0.com/signup
+
+<Tabs tabs={["Auth Code", "Implicit Flow", "Auth Code w/ PKCE"]}>
+
+<Tab>
+
+<SnackInline label='Auth0 Auth Code' dependencies={['expo-auth-session', 'expo-random', 'expo-web-browser']}>
+
+<!-- prettier-ignore -->
+```tsx
+import React, { useState, useEffect } from 'react';
+import { makeRedirectUri, useAuthRequest, ResponseType, Prompt } from 'expo-auth-session';
+import * as WebBrowser from 'expo-web-browser';
+import { Button, Platform } from 'react-native';
+
+WebBrowser.maybeCompleteAuthSession();
+
+const discovery = {
+  authorizationEndpoint: 'https://<AUTH0_DOMAIN>/authorize',
+  tokenEndpoint: 'https://<AUTH0_DOMAIN>/oauth/token',
+};
+
+const App: React.FC = () => {
+  const useProxy = Platform.select({ web: false, default: true });
+  const [randomValue] = useState(Math.random().toString(36).substring(2, 15));
```

Comment:
This is insecure, maybe try using `import { generateHexStringAsync } from 'expo-auth-session/build/PKCE'`, could be a good PR to expose that as a hook in the API.